modules/distribution-workflowoperation/src/main/java/org/opencastproject/workflow/handler/distribution/PublishEngageWorkflowOperationHandler.java

```diff
@@ -693,14 +702,20 @@ protected MediaPackage merge(MediaPackage mediaPackageForSearch) throws Workflow
    *          the mediapackage that is currently published
    * @return the merged mediapackage
    */
-  protected MediaPackage mergePackages(MediaPackage updatedMp, MediaPackage publishedMp) {
+  protected MediaPackage mergePackages(MediaPackage updatedMp, MediaPackage publishedMp,
+          List<MediaPackageElementFlavor> forceFlavors) {
     if (publishedMp == null)
       return updatedMp;
 
     MediaPackage mergedMediaPackage = (MediaPackage) updatedMp.clone();
     for (MediaPackageElement element : publishedMp.elements()) {
       String type = element.getElementType().toString().toLowerCase();
       if (updatedMp.getElementsByFlavor(element.getFlavor()).length == 0) {
+        if (forceFlavors.stream().anyMatch((f) -> element.getFlavor().matches(f))) {
+          logger.info("Forcing update of {} {} due to its flavor {}", type, element.getIdentifier(),
+                  element.getFlavor().toString());
+          continue;
+        }
```

Comment:
@lkiesow Can you clarify in the log message that this is "Forcing removal of {} {} due to ...." instead of the "Forcing update". The continue here omits this existing published element from being part of the merged updated mediapackage.

Reply:
Makes sense. I've tried to clarify that the update is a removal. The message should now read somewhat like this:

> Forcing removal of attachment 153eff86-1578-4567-a8a4-174d915c2f94 due to the absence of a new element with flavor security/series+xacml